contracts/crowdsale/validation/IndividuallyCappedCrowdsale.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^ 0.4.18;
+
+import "../../math/SafeMath.sol";
+import "../Crowdsale.sol";
+import "../../ownership/Ownable.sol";
+
+/**
+ * @dev Crowdsale with per-user caps
+ */
+contract IndividuallyCappedCrowdsale is Crowdsale, Ownable {
+  using SafeMath for uint256;
+
+  mapping(address => uint256) public contributions;
+  mapping(address => uint256) public caps;
+
+  /**
+   * @param _beneficiary Address to be capped
+   * @param _cap Wei limit for individual contribution
+   */
+  function setUserCap(address _beneficiary, uint256 _cap) external onlyOwner {
```

Comment:
Should we consider adding another function accepting an array of addresses? This is an implicit whitelist, it can be expensive to whitelist each crowdsale address in a different function call.